src/cqueues.c

```diff
@@ -1982,10 +1982,22 @@ static cqs_status_t cqueue_resume(lua_State *L, struct cqueue *Q, struct callinf
 		}
 	} else {
 		nargs = lua_gettop(T->L);
-		if (status != LUA_YIELD) {
+		if (status == LUA_OK && lua_getstack(T->L, 0, &(lua_Debug){}) > 0) {
```

Comment:
> I think the test should be status == LUA_OK && lua_getstack(...) <= 0, or even better status == LUA_OK && lua_getstack(...) != 1

Why? does >0 vs <= 0 matter?
==> What I have is the same as what the lua coroutine library has. 

This branch is what to do if the coroutine is already running; e.g.:
```lua
local cq = cqueues.new()
local co = coroutine.create(function()
    print(cq:step())
end)
cq:attach(co)
coroutine.resume(co)
```

> If we've been given a dead coroutine or a running coroutine then lua_resume() will return a descriptive error, right?

A running coroutine, yes.
However, a dead coroutine is not handled well. Note that lua itself checks for this: http://www.lua.org/source/5.3/lcorolib.c.html#auxresume